torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -67,32 +83,34 @@ ProfilingGraphExecutorImpl::ProfilingGraphExecutorImpl(
 ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(Stack& stack) {
   std::lock_guard<std::mutex> lock(compile_mutex);
   GRAPH_DEBUG("Running ProfilingGraphExecutorImpl ", this);
+
```

Comment:
Can we separate getPlanFor into one function that does the mutable changes to the executors state, and others that are pure functions that run various optimization phase? I am having trouble following what things need to get set, and how the flags interact with this.

Reply:
fixed